File: backend/src/main/java/com/yagubogu/member/dto/MemberInfoResponse.java
package com.yagubogu.member.dto;

import com.yagubogu.member.domain.Member;
import java.time.LocalDate;

public record MemberInfoResponse(
        String nickname,
        LocalDate createdAt,
        String favoriteTeam,
        String profileImageUrl
) {

    public static MemberInfoResponse from(final Member member) {
        return new MemberInfoResponse(
                member.getNickname(),
                member.getCreatedAt().toLocalDate(),
                member.getTeam().getShortName(),
                member.getImageUrl()
        );
    }

    public static MemberInfoResponse fromNullableTeam(final Member member) {
        return new MemberInfoResponse(
                member.getNickname(),
                member.getCreatedAt().toLocalDate(),
                null,
                member.getImageUrl()
        );
    }
}

File: android/app/src/main/java/com/yagubogu/data/dto/response/member/MemberInfoResponse.kt
package com.yagubogu.data.dto.response.member

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class MemberInfoResponse(
    @SerialName("nickname")
    val nickname: String, // 닉네임
    @SerialName("createdAt")
    val createdAt: String, // 가입일
    @SerialName("favoriteTeam")
    val favoriteTeam: String, // 나의 팀
    @SerialName("profileImageUrl")
    val profileImageUrl: String, // 프로필 이미지
)
